OpenAPIService/PowershellFormatter.cs

```diff
@@ -189,5 +192,41 @@ private static void ResolveFunctionParameters(OpenApiOperation operation)
                 }
             }
         }
+
+        /// <summary>
+        /// Singularizes and deduplicates segment names in an operation id.
+        /// </summary>
+        /// <param name="operationId">The target operationId value.</param>
+        /// <returns>The operationId with the segments singularized and deduplicated.</returns>
+        private static string SingularizeAndDeduplicateOperationId(string operationId)
+        {
+            if (string.IsNullOrEmpty(operationId))
+                return operationId;
+
+            // Plurality is assumed to be words ending in 's'/'S'.
+            char[] plurals = { 's', 'S' };
+
+            var segments = operationId.Split('.').ToList();
+
+            // The last segment is ignored as a rule.
+            for (int x = segments.Count - 2; x >= 0; x--)
+            {
+                var segment = segments[x].TrimEnd(plurals);
```

Comment:
```suggestion
                var segment = segments[x].Singularize();
```
Can we use [Humanizer.NET](https://humanizr.net/#singularize) to handle singularization? The library will handle singularization of English words.

Reply:
Update as
```CSharp
Singularize(inputIsKnownToBePlural: false)
```
To cater for words like: `axis`, `class`